[easy] Add missing type annotation

ttung opened the pull request:
### Description
Add one missing type annotation that improves quality of life. :)

### Test plan
definitely YOLO.

phoenixAja commented on src/py/aspen/database/connection.py:
```diff
@@ -32,7 +32,7 @@ def init_db(db_uri: str) -> SqlAlchemyInterface:
 
 @contextmanager
-def session_scope(interface: SqlAlchemyInterface):
+def session_scope(interface: SqlAlchemyInterface) -> Iterator[Session]:
```
when should we use Iterator vs Generator for the type hints? Generator as the type hint here seems a little more specific to me.

ttung replied:
Switched to `Generator`; I agree that it makes more sense.

Commit:
[easy] Add missing type annotation

### Description
Add one missing type annotation that improves quality of life. :)

### Test plan
definitely YOLO.
